src/main/java/org/reactome/server/util/DoiPathwayCache.java

```diff
@@ -0,0 +1,41 @@
+package org.reactome.server.util;
+
+import org.reactome.server.graph.domain.result.PathwayResult;
+import org.reactome.server.graph.service.DoiService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.Collection;
+
+@Component
+public class DoiPathwayCache {
+    private static final Logger infoLogger = LoggerFactory.getLogger("infoLogger");
+
+    private DoiService doiService;
+
+    private Collection<PathwayResult> doiPathways;
+
+    @Autowired
+    public void setDoiService(DoiService doiService) {
+        this.doiService = doiService;
+    }
+
+    @PostConstruct
+    public void getDoiPathwayResult() {
+
+        try {
+            if (doiPathways == null) {
+                doiPathways = doiService.getAllDoiPathway();
+            }
+        } catch (NullPointerException e) {
```

Comment:
It is not recommended to catch NPE
https://stackoverflow.com/questions/15146339/catching-nullpointerexception-in-java
why are you catching it ?

Reply:
will change it